### PR TITLE
Net plugin logging 📦

### PR DESCRIPTION
**Pull request by heifner**
## Change Description

- Modify `peer_[x]log`  to not require a mutex lock.
-  `peer_[x]log` now only valid on connection strand (checked and enforced).
- Cleanup handling of peer name and remove unneeded locks for strand data
- Fixes an invalid call to strand specific method for pruned blocks

## Change Type
**Select *ONE*:**
- [ ] Documentation

- [x] Stability bug fix

- [x] Other

- [ ] Other - special case

## Testing Changes
**Select *ANY* that apply:**
- [ ] New Tests

- [ ] Existing Tests

- [ ] Test Framework

- [ ] CI System

- [ ] Other

## Consensus Changes
- [ ] Consensus Changes

## API Changes
- [ ] API Changes

## Documentation Additions
- [ ] Documentation Additions

### CODE REVIEW

File: plugins/net_plugin/net_plugin.cpp [hunk 1]
```diff
@@ -779,12 +788,12 @@ namespace eosio {
 
       fc::variant_object get_logger_variant()  {
          fc::mutable_variant_object mvo;
-         mvo( "_name", peer_name());
-         std::lock_guard<std::mutex> g_conn( conn_mtx );
-         mvo( "_id", conn_node_id )
-            ( "_sid", conn_node_id.str().substr( 0, 7 ) )
-            ( "_ip", remote_endpoint_ip )
-            ( "_port", remote_endpoint_port )
+         mvo( "_name", log_p2p_address)
```
Comment (dimas1185) [inline plugins/net_plugin/net_plugin.cpp:791]: I don't see `log_p2p_address` is assigned somewhere

Comment (heifner) [reply plugins/net_plugin/net_plugin.cpp:791]: Nice catch!

Comment (heifner) [reply plugins/net_plugin/net_plugin.cpp:791]: Pushed fix.

File: plugins/net_plugin/net_plugin.cpp [hunk 2]
```diff
@@ -1663,8 +1669,8 @@ namespace eosio {
                ("r", sync_last_requested_num)("e", sync_next_expected_num)("k", sync_known_lib_num)("s", sync_req_span) );
 
       if( fork_head_block_num < sync_last_requested_num && sync_source && sync_source->current() ) {
-         fc_ilog( logger, "ignoring request, head is ${h} last req = ${r} source is ${p}",
-                  ("h", fork_head_block_num)( "r", sync_last_requested_num )( "p", sync_source->peer_name() ) );
+         fc_ilog( logger, "ignoring request, head is ${h} last req = ${r}",
```
Comment (dimas1185) [inline plugins/net_plugin/net_plugin.cpp]: was there an intention to call peer_* macro or use connection_id instead?

**Review by dimas1185 — APPROVED** [review]

File: plugins/net_plugin/net_plugin.cpp (hunk 2 above)
Comment (heifner) [reply plugins/net_plugin/net_plugin.cpp]: Could add in `sync_source->connection_id` wouldn't hurt

**Pull request merged**